### PR TITLE
Add PR test workflow and check-in more testcases

### PR DESCRIPTION
**Pull request by zhulinJulia24**
- refactor testcase
  - change cases into dir
  - model coverage config in yml
- add more testcase
  - add kvint8 quantization
  - add pipeline interface testcase
- add pr test case and workflow

### CODE REVIEW

File: .github/workflows/pr_ete_test.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+      - "cmake/**"
+      - "src/**"
+      - "autotest/**"
+      - "3rdparty/**"
+      - "lmdeploy/**"
+      - "requirements/**"
+      - "requirements.txt"
+      - "CMakeLists.txt"
+      - "setup.py"
+  push:
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:16]: could remove push

File: .github/workflows/pr_ete_test.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    branches:
+      - main
+    paths:
+      - ".github/workflows/pr_ete_test.yml"
+      - "cmake/**"
+      - "src/**"
+      - "autotest/**"
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:23]: could remove `autotest`

File: .github/workflows/pr_ete_test.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      - "3rdparty/**"
+      - "lmdeploy/**"
+      - "requirements/**"
+      - "requirements.txt"
+      - "CMakeLists.txt"
+      - "setup.py"
+    tags:
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:30]: remove `tags`  section

File: .github/workflows/pr_ete_test.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  workflow_dispatch:
+
+
+env:
+  HOST_PIP_CACHE_DIR: /nvme/github-actions/pip-cache
+  HOST_LOCALTIME: /usr/share/zoneinfo/Asia/Shanghai
+
+
+jobs:
+  test_functions:
```
Comment (RunningLeon) [inline .github/workflows/pr_ete_test.yml:25]: may change the job name

**Review by RunningLeon — APPROVED** [review]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**